pytradfri/api/aiocoap_api.py

```diff
@@ -175,11 +175,10 @@ async def request(self, api_commands):
 
     async def _observe(self, api_command):
         """Observe an endpoint."""
-        duration = api_command.observe_duration
         url = api_command.url(self._host)
         err_callback = api_command.err_callback
 
-        msg = Message(code=Code.GET, uri=url, observe=duration)
+        msg = Message(code=Code.GET, uri=url, observe=0)
```

Comment:
We're now ignoring the observe duration set on the command. Shouldn't we remove that attribute too? It's already a breaking change by ignoring it.

Reply:
Yes, I kind of agree that it seems like it would be better to remove duration (observe_duration) completely. But for now, I'm proposing this smaller and safer change.

It seems like this use of duration for aiocoap observe has been wrong since it was introduced in 482dd897. It seems like setting duration to anything but 0 never worked. This change can thus not break anything - only improve things. It doesn't really change the API and I wouldn't call it a breaking change.

The duration parameter is however used for libcoap where it works. But as mentioned in the PR description, I think that could be seen as a misfeature.

Reply:
What is happening when setting the duration and why doesn't it work as expected?

Reply:
It was putting the duration value in a completely different header field where the only valid values are 0 and 1, and where 0 is the only value that is correct and actually works. When the bogus value is put there, observe doesn't work.

For example, look for `opt.observe == 0` in `aiocoap/protocol.py`.